--- xosview-1.8.1/linux/netmeter.cc.orig	2003-10-20 02:51:51.000000000 +0200
+++ xosview-1.8.1/linux/netmeter.cc	2003-10-26 23:15:09.035652672 +0100
@@ -43,6 +43,7 @@
 #include <errno.h>
 #include <iostream>
 #include <iomanip>
+#include <string>
 
 NetMeter::NetMeter( XOSView *parent, float max )
   : FieldMeterGraph( parent, 3, "NET", "IN/OUT/IDLE" ){
@@ -145,6 +146,7 @@
         return;
         }
 
+    std::string str_in;
     unsigned long in, out, ig;
     unsigned long totin = 0, totout = 0;
     char buf[1024];
@@ -174,6 +176,10 @@
 	  while (ifs)
 	      {
 	      ifs.ignore(1024, ':');
+	      ifs >> str_in;
+	      if (str_in == "No")
+		  continue;
+	      in = strtoul(str_in.c_str(), NULL, 10);
 	      ifs >> in >> ig >> ig >> ig >> ig >> ig >> ig >> ig >> out;
 
 	      if (!ifs.eof())
