--- xosview-1.8.4/xwin.cc.orig	2015-06-04 16:49:00.000000000 +0200
+++ xosview-1.8.4/xwin.cc	2024-02-04 12:46:57.721543774 +0100
@@ -418,8 +418,6 @@ const char *XWin::getResource( const cha
   {
     std::cerr << "Error:  Couldn't find '" << name << "' resource in the resource database!\n";
     exit (-1);
-      /*  Some compilers aren't smart enough to know that exit() exits.  */
-    return '\0';
   }
 }
 
--- xosview-1.9.4/linux/acpitemp.cc.orig	2012-07-09 17:56:42.000000000 +0200
+++ xosview-1.9.4/linux/acpitemp.cc	2024-02-04 16:31:49.978449947 +0100
@@ -108,12 +108,12 @@ void ACPITemp::getacpitemp( void ) {
   std::ifstream high_file(_highfile);
 
   if (!temp_file) {
-    std::cerr << "Can not open file : " << temp_file << std::endl;
+    std::cerr << "Can not open file : " << _tempfile << std::endl;
     parent_->done(1);
     return;
   }
   if (!high_file) {
-    std::cerr << "Can not open file : " << high_file << std::endl;
+    std::cerr << "Can not open file : " << _highfile << std::endl;
     parent_->done(1);
     return;
   }
