diff -ur xosview-1.7.3.orig/config/Makefile.config.in xosview-1.7.3/config/Makefile.config.in
--- xosview-1.7.3.orig/config/Makefile.config.in	Wed Nov 17 07:33:25 1999
+++ xosview-1.7.3/config/Makefile.config.in	Wed Jul 12 16:02:01 2000
@@ -22,7 +22,7 @@
 #-----------------------------------------------------------------
 
 # Optimized CXXFLAGS
-CXXFLAGS      += $(CFLAGS) -Wall -O4 -pipe -I@x_includes@ \
+CXXFLAGS      += $(CFLAGS) -Wall -pipe -I@x_includes@ \
                 -DXOSVIEW_VERSION=\"$(VERS)\" -I$(TOP) -I$(TOP)/@host_dir@ @DEFS@
 # DEBUG CXXFLAGS
 #CXXFLAGS      += $(CFLAGS) -Wall -ggdb -pipe -I@x_includes@ \
diff -ur xosview-1.7.3.orig/config/Makefile.linux.in xosview-1.7.3/config/Makefile.linux.in
--- xosview-1.7.3.orig/config/Makefile.linux.in	Fri Nov 12 06:13:22 1999
+++ xosview-1.7.3/config/Makefile.linux.in	Wed Jul 12 16:02:01 2000
@@ -22,12 +22,6 @@
 raidmeter.o \
 MeterMaker.o
 
-ifeq (@host_cpu@, m68k)
-# nothing else for m68k
-else
-OBJS += serialmeter.o
-endif
-
 CFILES := $(OBJS:.o=.cc)
 DEPFILES := $(OBJS:%=.%.d)
 
diff -ur xosview-1.7.3.orig/linux/MeterMaker.cc xosview-1.7.3/linux/MeterMaker.cc
--- xosview-1.7.3.orig/linux/MeterMaker.cc	Fri Nov 12 06:13:22 1999
+++ xosview-1.7.3/linux/MeterMaker.cc	Wed Jul 12 16:04:16 2000
@@ -15,7 +15,6 @@
 #include "pagemeter.h"
 #include "netmeter.h"
 #include "intmeter.h"
-#include "serialmeter.h"
 #include "loadmeter.h"
 #include "btrymeter.h"
 #include "diskmeter.h"
@@ -54,14 +53,6 @@
 
   if (_xos->isResourceTrue("disk"))
       push(new DiskMeter(_xos, atof(_xos->getResource("diskBandwidth"))));
-
-  // check for the serial meters.
-#ifndef __mc68000__
-  for (int i = 0 ; i < SerialMeter::numDevices() ; i++)
-    if (_xos->isResourceTrue(SerialMeter::getResourceName(
-      (SerialMeter::Device)i)))
-        push(new SerialMeter(_xos, (SerialMeter::Device)i));
-#endif
 
   // check for the interrupt meter
   if (_xos->isResourceTrue("interrupts")) {
