--- xosview-1.7.3/config/configure.in	Fri Jan 26 16:31:32 2001
+++ xosview-1.7.3/config/configure.in.fix	Fri Jan 26 16:31:57 2001
@@ -11,7 +11,8 @@
 AC_PATH_X
 CXXFLAGS="$CXXFLAGS -I$x_includes"
 LIBS="-L$x_libraries -R$x_libraries $LIBS"
-ELF_LINK_FLAGS="-Wl,-R$x_libraries"
+#ELF_LINK_FLAGS="-Wl,-R$x_libraries"
+ELF_LINK_FLAGS="-Wl"
 dnl AC_PREFIX_DEFAULT(/usr/X11)	dnl No longer needed with our
 dnl 				x_includes magic in Makefile.top.in.
 dnl  Check for /usr/X11R6.  If it exists, make that the default prefix.
