diff -ur xosview-1.8.0.orig/config/Makefile.linux.in xosview-1.8.0/config/Makefile.linux.in
--- xosview-1.8.0.orig/config/Makefile.linux.in	Sun Feb 24 23:00:47 2002
+++ xosview-1.8.0/config/Makefile.linux.in	Fri Oct 11 00:27:54 2002
@@ -23,7 +23,7 @@
 lmstemp.o \
 MeterMaker.o
 
-FAIL_ARCHS =3D m68k sparc
+FAIL_ARCHS = alpha m68k powerpc sparc
 ifneq (,$(findstring @host_cpu@, $(FAIL_ARCHS)))
 # nothing else for m68k
 else
diff -ur xosview-1.8.0.orig/linux/MeterMaker.cc xosview-1.8.0/linux/MeterMaker.cc
--- xosview-1.8.0.orig/linux/MeterMaker.cc	Sun Feb 24 23:00:47 2002
+++ xosview-1.8.0/linux/MeterMaker.cc	Fri Oct 11 00:18:11 2002
@@ -57,7 +57,7 @@
       push(new DiskMeter(_xos, atof(_xos->getResource("diskBandwidth"))));
 
   // check for the serial meters.
-#ifndef __mc68000__
+#if !defined(__alpha__) && !defined(__mc68000__) && !defined(__powerpc__) && !defined(__sparc__)
   for (int i = 0 ; i < SerialMeter::numDevices() ; i++)
     if (_xos->isResourceTrue(SerialMeter::getResourceName(
       (SerialMeter::Device)i)))
