--- xosview-1.8.1/linux/serialmeter.cc.orig	2003-10-20 03:20:42.000000000 +0200
+++ xosview-1.8.1/linux/serialmeter.cc	2003-10-26 22:54:13.653499768 +0100
@@ -35,12 +35,12 @@
 #if !defined(__hppa__) && !defined(__mips__)
 #include <sys/io.h>
 #endif
-#if !defined(__alpha__) && !defined(__ia64__) && !defined(__hppa__) && !defined(__arm__) && !defined(__mips__)
+#if !defined(__alpha__) && !defined(__ia64__) && !defined(__hppa__) && !defined(__arm__) && !defined(__mips__) && !defined(__sparc__)
 #include <sys/perm.h>
 #define HAVE_IOPERM
 #endif
 #else
-#ifndef __alpha__
+#if !defined(__alpha__) && !defined(__sparc__)
 #include <asm/io.h>
 #endif
 #endif
