--- xosview-1.7.3/linux/serialmeter.cc.orig	Tue Dec 18 17:03:41 2001
+++ xosview-1.7.3/linux/serialmeter.cc	Tue Dec 18 17:05:24 2001
@@ -140,6 +140,7 @@
   }
   else { // Use user specified port base.
     istrstream istrm(res);
+    unsigned short int hex;
     unsigned short int tmp = 0;
     istrm >> hex >> tmp;
     return tmp;
--- xosview-1.7.3/xwin.h.orig	Tue Dec 18 16:33:55 2001
+++ xosview-1.7.3/xwin.h	Tue Dec 18 16:34:06 2001
@@ -107,7 +107,7 @@
     Event( XWin *parent, int event, EventCallBack callBack );
     virtual ~Event( void ){}
 
-    friend XWin;
+    friend class XWin;
 
     void callBack( XEvent &event )
       { if ( event.type == event_ ) (parent_->*callBack_)( event ); }
